### PR TITLE
Doc: Clarify table properties on metadata metrics inferred column defaults

### PR DESCRIPTION
**Pull request by dramaticlly**
Based on comment from 
https://github.com/apache/iceberg/pull/5215/files#r915475413
and https://github.com/apache/iceberg/pull/5215/files#r915276052

100 as default limit is counted only toward top level columns, which means stats are also collected for nested children of top level columns, but not counted in limit.

Example:
```
root
 |-- foo1: struct (nullable = true)
 |    |-- bar1: integer (nullable = false)
 |    |-- baz1: string (nullable = true)
 |-- foo2: struct (nullable = true)
 |    |-- bar2: integer (nullable = false)
 |    |-- baz2: string (nullable = true)
...
 |-- foo101: struct (nullable = true)
 |    |-- bar101: integer (nullable = true)
 |    |-- baz101: string (nullable = true)
```

- stats for `foo1.bar1`, `foo1.baz1`, `foo2.bar2`, `foo2.baz2` will all be collected use default (truncate(16))
- stats for `foo101.bar101`, `foo101.baz101` will be none

CC @stevenzwu

### CODE REVIEW

**Review by stevenzwu — APPROVED** [review]

**Comment by stevenzwu** [comment]
thanks @dramaticlly for clarifying the doc. I have always thought the 100 limit is applicable to nested fields. didn't know it is limited to the top level.

**Comment by dramaticlly** [comment]
@nastra @amogh-jahagirdar can you help review this as well?

File: docs/docs/configuration.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ Iceberg tables support table properties to configure table behavior, like the de
...
 | write.location-provider.impl                         | null                        | Optional custom implementation for LocationProvider                                                                                                                                               |
 | write.metadata.compression-codec                     | none                        | Metadata compression codec; none or gzip                                                                                                                                                          |
-| write.metadata.metrics.max-inferred-column-defaults  | 100                         | Defines the maximum number of columns for which metrics are collected                                                                                                                             |
+| write.metadata.metrics.max-inferred-column-defaults  | 100                         | Defines the maximum number of top level columns for which metrics are collected. Number of stored metrics can be higher than this limit for table with nested schema                                         |
```
Comment (nastra) [inline docs/docs/configuration.md:64]:
```suggestion
| write.metadata.metrics.max-inferred-column-defaults  | 100                         | Defines the maximum number of top level columns for which metrics are collected. Number of stored metrics can be higher than this limit for a table with nested fields                                         |
```

**Review by nastra — APPROVED** [review]

File: docs/docs/configuration.md [hunk 2]
```diff
@@ -61,7 +61,7 @@ Iceberg tables support table properties to configure table behavior, like the de
...
 | write.location-provider.impl                         | null                        | Optional custom implementation for LocationProvider                                                                                                                                               |
 | write.metadata.compression-codec                     | none                        | Metadata compression codec; none or gzip                                                                                                                                                          |
-| write.metadata.metrics.max-inferred-column-defaults  | 100                         | Defines the maximum number of columns for which metrics are collected                                                                                                                             |
+| write.metadata.metrics.max-inferred-column-defaults  | 100                         | Defines the maximum number of top level columns for which metrics are collected. Number of stored metrics can be higher than this limit for a table with nested fields                                         |
```
Comment (nastra) [inline docs/docs/configuration.md:64]:
```suggestion
| write.metadata.metrics.max-inferred-column-defaults  | 100                         | Defines the maximum number of top level columns for which metrics are collected. Number of stored metrics can be higher than this limit for a table with nested fields                            |
```

**Comment by dramaticlly** [comment]
Thank you @nastra and @stevenzwu for the review

**Pull request merged**